openmls/src/group/core_group/new_from_welcome.rs

```diff
@@ -182,12 +217,14 @@ impl CoreGroup {
         key_package: &KeyPackage,
         welcome_secrets: &[EncryptedGroupSecrets],
         backend: &impl OpenMlsCryptoProvider,
-    ) -> Result<Option<EncryptedGroupSecrets>, KeyPackageError> {
+    ) -> Option<EncryptedGroupSecrets> {
```

Comment:
I know we make our lives easier by getting rid of a `Result`, but at the same time we're making debugging harder, because we "silently" return `None` instead of surfacing an error when something goes wrong. I did this as well in a recent PR, but I thought about it and I don't think this is a good idea anymore.

Reply:
I think we discussed this a couple times now. There are two reasons not to return errors in functions like this.
- There's no point in returning an error if it's nothing the caller can act upon.
- The idiomatic way is to return an `Option` for lookup functions. The main reason for this is again the first one. The caller wanted to get a value. The internals of the mechanisms are irrelevant.

Reply:
Then at least we should add some logging in case something went wrong during the lookup, or maybe a `debug_assert`. I understand that `Option` makes sense for lookup functions. But as a caller, I'd expect `None` if the value isn't there, not if there is something else that happened that prevented the lookup from happening in the first place. To me, it hints to the fact that this is more than a lookup function.

Reply:
Or looking at it in another way: If the caller gets a `LibraryError`, they at least know there's nothing they can do about that and they can decide to handle it in some way. If the caller gets a `None`, they can decide to do something different. However, if there is a `None` in both cases, the caller has no way of knowing if something unexpected went wrong or if the value is not there due to something else that they may have done.